### ISSUE TITLE
How to update headers in StableOtlpHTTPMetricExporter?

### ISSUE THREAD

**Issue by JustasL** [open-telemetry/opentelemetry-swift#650]
Hello, is there are way to update `StableOtlpHTTPMetricExporter` headers after it has been setup?

We want to provide authorisation token, which can expire, and will need to be updated in the future
But at the moment it seems headers can only be setup during init?

**Comment by n-ravichandran** [comment]
I am unclear about their recommended approach for including the auth token in the header while using their exporter. Their exporter includes a `URLSession` parameter, so you could accomplish this by supplying a `URLSession` object that employs a custom `URLProtocol` to intercept requests. Within the interceptor protocol class, you can alter the request header to include the auth token. If your auth token has expired, you can also perform a token refresh there and continue the request with the new token.

**Comment by JustasL** [comment]
Thanks, I'll give it a try, but it looks a little bit overcomplicated solution to simple issue

**Comment by bryce-b** [comment]
@JustasL 
You can pass headers to any of the Exporters through the [OtlpConfiguration](https://github.com/open-telemetry/opentelemetry-swift/blob/a0261facbc0cada5fa796fc48c8f0ec821f2c000/Sources/Exporters/OpenTelemetryProtocolCommon/common/OtlpConfiguration.swift#L30) object as the `config` parameter in the initializer method.

**Comment by JustasL** [comment]
@bryce-b yeah, but it is constant, I can not update headers after the token has expired and has been updated

**Comment by bryce-b** [comment]
Would it be possible to re-create the exporter when you need to refresh the headers?

**Comment by bryce-b** [comment]
@JustasL I'm having trouble finding where android provides a means to update the headers, could you share it?

**Comment by bryce-b** [comment]
refined issue: https://github.com/open-telemetry/opentelemetry-swift/issues/651

**Comment by JustasL** [comment]
My android colleague is using [this method](https://github.com/search?q=repo%3Aopen-telemetry%2Fopentelemetry-java%20setHeaders&type=code)

**Comment by bryce-b** [comment]
I'm going to close this issue in favor of conversing in #651
